web/app/scripts/views/record/list-controller.js

```diff
@@ -38,7 +38,7 @@
             return RecordSchemaState.get(currentSchemaId)
                 .then(function(recordSchema) {
                     ctl.recordSchema = recordSchema;
-                });
+                }).then(onSchemaLoaded);
```

Comment:
Andrew pointed out on a PR of mine last week that promises behave differently depending on whether or not they're returning something.  Apparently this setup, where the `.then()` function doesn't return anything, results in that code and the next `.then()` in the chain running concurrently (or at least not being guaranteed to run sequentially).  Probably the dynamics of this setup are such that the problem would never actually arise, but I think it would be more correct/airtight to have the first `then` return recordSchema (still setting it on the controller, just also returning it) and have onSchemaLoaded accept it as an argument.